bitcoin-rpc-provider/src/lib.rs

```diff
@@ -143,6 +145,27 @@ impl Wallet for BitcoinCoreProvider {
         Ok(pk.key)
     }
 
+    fn sign_for_witness<C: Signing>(
+        &self,
+        secp: &Secp256k1<C>,
+        fund: &Transaction,
```

Comment:
nit: might be better to name that something more generic in case it gets reused somewhere else.